### DISCUSSION TITLE
Is there a way to remove a particular entity from a knowledge base given their entity id?

### DISCUSSION THREAD

**Discussion by epb378** [explosion/spaCy#8307] [Help: Best practices]
## Spacy knowledge base

Is there an elegant way to remove an entity from a knowledge base? It looks like you might be able to copy all the existing entities (minus the one to be deleted) to a new kb, but this doesn't seem like an elegant approach, and it doesn't look like it will keep the probabilities of each alias being assigned to each entity.
[source](https://github.com/explosion/spaCy/blob/master/spacy/kb.pyx)

**Comment by svlandeg** [comment]
There's currently not really a method to delete one entity, as it was kind of assumed that a KB is built once and isn't dynamically being changed. Might I ask what's your use-case for removing an entity after construction of the KB?

And yes you're right that if you'd copy the entities over to a new KB, you'd also have to specify the aliases again for each.

**Reply by epb378** [reply]
Interesting...Thanks for the quick response. The main use case is to handle errors. If a name is entered incorrectly it would be great to remove them without having to do it all again.
